.buildkite/heartbeat/heartbeat-pipeline.yml

```diff
@@ -110,11 +121,13 @@ steps:
   - group: "Heartbeat Extended Testing MacOS"
     key: "heartbeat-extended-tests-macos"
     if: build.env("BUILDKITE_PULL_REQUEST") == "false" || build.env("GITHUB_PR_LABELS") =~ /.*macOS.*/
+
     steps:
       - label: ":mac: Heartbeat MacOS Unit Tests"
         key: "macos-extended"
-        if: build.env("GITHUB_PR_TRIGGER_COMMENT") == "heartbeat for macos" || build.env("GITHUB_PR_LABELS") =~ /.*macOS.*/
-        command: "cd ${BEATS_PROJECT_NAME} && mage build unitTest"
+        command: |
+          cd heartbeat
+          mage build unitTest
```

Comment:
if we change this to:

```
          set -euo pipefail
          source .buildkite/scripts/install_macos_tools.sh
          cd heartbeat
          mage build unitTest
```

we should be able to remove the conditional in the [pre-command hook](https://github.com/elastic/beats/blob/main/.buildkite/hooks/pre-command#L14).